crates/blockchain-tree-api/src/error.rs

```diff
@@ -339,15 +339,18 @@ impl InsertBlockErrorKindTwo {
     ///
     /// This is intended to be used to determine if the block should be marked as invalid.
     #[allow(clippy::match_same_arms)]
-    pub fn is_invalid_block(self) -> Result<(), InsertBlockFatalError> {
+    pub fn is_invalid_block(self) -> Result<InsertBlockValidationError, InsertBlockFatalError> {
```

Comment:
perhaps we could find a more appropriate function name here, like `ensure_validation_error`

Reply:
yeah agree, let me think of something

Reply:
couldnt think of anything better, just went with `ensure_validation_error`